### PR TITLE
fix(client): add error handling in client

### PR DESCRIPTION
**Pull request by hwdef**
Signed-off-by: hwdef <hwdef97@gmail.com>

### Ⅰ. Describe what this PR did
add error handling in client

### Ⅱ. Does this pull request fix one issue?

NONE

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)

may be need test

### CODE REVIEW

File: client/request.go [hunk 1]
```diff
@@ -104,8 +104,14 @@ func (client *APIClient) hijack(ctx context.Context, path string, query url.Valu
 	}
 
 	if tcpConn, ok := conn.(*net.TCPConn); ok {
-		tcpConn.SetKeepAlive(true)
-		tcpConn.SetKeepAlivePeriod(30 * time.Second)
+		err = tcpConn.SetKeepAlive(true)
```
Comment (yeya24) [inline client/request.go]: I hope we can use the format of 
``` Go
if err := tcpConn.SetKeepAlive(); err != nil {
    return ...
}
```
Please address these all

Comment (hwdef) [reply client/request.go]: @yeya24 
thanks for your suggestion.I changed it.

File: client/utils.go [hunk 2]
```diff
@@ -31,10 +31,17 @@ func decodeBody(obj interface{}, body io.Reader) error {
...
 
-func ensureCloseReader(resp *Response) {
+func ensureCloseReader(resp *Response) error {
 	if resp != nil && resp.Body != nil {
 		// close body ReadCloser to make Transport reuse the connection
-		io.CopyN(ioutil.Discard, resp.Body, 512)
-		resp.Body.Close()
+
+		if _, err := io.CopyN(ioutil.Discard, resp.Body, 512); err != nil {
+			return fmt.Errorf("failed to ensure close reader: %v", err)
```
Comment (Starnop) [inline client/utils.go]: IMOP, `return err` is enough. In addition, the caller should also handle the error that this function return.

File: client/utils.go [hunk 3]
```diff
@@ -31,10 +31,15 @@ func decodeBody(obj interface{}, body io.Reader) error {
...
-func ensureCloseReader(resp *Response) {
+func ensureCloseReader(resp *Response) error {
 	if resp != nil && resp.Body != nil {
 		// close body ReadCloser to make Transport reuse the connection
-		io.CopyN(ioutil.Discard, resp.Body, 512)
-		resp.Body.Close()
+		if _, err := io.CopyN(ioutil.Discard, resp.Body, 512); err != nil {
+			return err
+		}
+		if err := resp.Body.Close(); err != nil {
```
Comment (Starnop) [inline client/utils.go]: `resp.Body.Close()` should be executed regardless of the result of `io.CopyN`

**Comment by Starnop** [comment]
LGTM.

**Pull request merged**